extensions/stackdriver/stackdriver.cc

```diff
@@ -85,15 +122,22 @@ bool StackdriverRootContext::onConfigure(
     logWarn("Unable to get plugin direction");
   }
 
-  auto exporter =
-      std::make_unique<ExporterImpl>(this, config_.test_logging_endpoint());
-  // logger takes ownership of exporter.
-  logger_ = std::make_unique<Logger>(local_node_info_, std::move(exporter));
+  if (!logger_) {
```

Comment:
Since we removed all variability from xds config and shoved it into bootstrap, this will work.
But we will have filters that need xds config and in that case this code needs to ensure that no only are things initialized but initialized with the same params. I think the fix is ok right now.

Reply:
`logger` is expensive and unnecessary to update, so we remove the variability here. But say if `logger` needs to be updated based on config, I think should we make callbacks safe by avoiding referencing the object that are up to destruction due to config update.

Reply:
Using `weak_ptr` is also a solution to this.